logical/framework/openapi.go

```diff
@@ -148,15 +148,19 @@ type OASMediaTypeObject struct {
 }
 
 type OASSchema struct {
-	Type        string                `json:"type,omitempty"`
-	Description string                `json:"description,omitempty"`
-	Properties  map[string]*OASSchema `json:"properties,omitempty"`
-	Items       *OASSchema            `json:"items,omitempty"`
-	Format      string                `json:"format,omitempty"`
-	Pattern     string                `json:"pattern,omitempty"`
-	Enum        []interface{}         `json:"enum,omitempty"`
-	Example     interface{}           `json:"example,omitempty"`
-	Deprecated  bool                  `json:"deprecated,omitempty"`
+	Type             string                `json:"type,omitempty"`
+	Description      string                `json:"description,omitempty"`
+	Properties       map[string]*OASSchema `json:"properties,omitempty"`
+	Items            *OASSchema            `json:"items,omitempty"`
+	Format           string                `json:"format,omitempty"`
+	Pattern          string                `json:"pattern,omitempty"`
+	Enum             []interface{}         `json:"enum,omitempty"`
+	Example          interface{}           `json:"example,omitempty"`
+	Deprecated       bool                  `json:"deprecated,omitempty"`
+	Required         bool                  `json:"required,omitempty"`
+	DisplayName      string                `json:"x-vault-display-name,omitempty" mapstructure:"x-vault-display-name,omitempty"`
```

Comment:
json usually uses `_` instead of `-`. Also why start with `x-`?

Reply:
`x-` is per the [extension spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#specificationExtensions) because these are not standard OpenAPI fields.

As for the `-` character in the rest of the field name, I'm open to picking a new standard for these. Looking over the OpenAPI spec, the standard fields are in camelCase, so for consistency within the output JSON I think that would be a good choice.

Reply:
Updated to camelCase.